pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -142,6 +142,7 @@
     public static final Pattern HEARTBEAT_NAMESPACE_PATTERN = Pattern.compile("pulsar/[^/]+/([^:]+:\\d+)");
     public static final Pattern SLA_NAMESPACE_PATTERN = Pattern.compile(SLA_NAMESPACE_PROPERTY + "/[^/]+/([^:]+:\\d+)");
     public static final String HEARTBEAT_NAMESPACE_FMT = "pulsar/%s/%s:%s";
+    public static final String HEARTBEAT_NAMESPACE_FMT_V2 = "pulsar/%s:%s";
```

Comment:
You also need to update the static variable `HEARTBEAT_NAMESPACE_PATTERN` and the method `checkHeartbeatNamespace` to support v2 version of the topic

Reply:
We already have  a new version of the method that does that.

```    public static String getHeartbeatNamespaceV2(String host, ServiceConfiguration config) {
        Integer port = null;
        if (config.getWebServicePort().isPresent()) {
            port = config.getWebServicePort().get();
        } else if (config.getWebServicePortTls().isPresent()) {
            port = config.getWebServicePortTls().get();
        }
        return String.format(HEARTBEAT_NAMESPACE_FMT_V2, host, port);
    }
    ```

Reply:
@aahmed-se I am talking about `checkHeartbeatNamespace` not `getHeartbeatNamespace`

Reply:
added it.